xdsl/dialects/riscv_scf.py

```diff
@@ -189,6 +186,32 @@ def parse(cls, parser: Parser) -> Self:
         return cls(lb, ub, step, iter_arg_operands, body)
 
 
+@irdl_op_definition
+class ForOp(ForRofBaseOp):
+    """
+    A for loop, counting up from lb to ub by step each iteration.
+    """
+    name = "riscv_scf.for"
+
+
+@irdl_op_definition
+class RofOp(ForRofBaseOp):
+    """
+    Reverse Order For loop.
+
+    MLIRs for loops have the implicit constraint that lb < ub, and step > 0.
+
+    In order to express loops that count down in higher-level IR, the rof op
+    is needed.
+
+    ROF has the semantics of going from lb to ub, decrementing by step each time.
+
+    In order to convert a for to a rof, one needs to switch lb and ub.
+    (for the normalized case that (ub - lb) % step == 0)
+    """
+    name = "riscv_scf.rof"
```

Comment:
I'm not sure I like this, I think for the custom syntax especially it's confusing. I'd say we want the new custom syntax to be 
`riscv_scf.rof %i : !riscv.reg<> = %ub down to %lb by %step ... {`
The `down to` and switching of ub and lb will hopefully be a little clearer to the reader. I would say that the step still needs to be >0 for both loops.

Reply:
Makes sense!

Reply:
But that's just commentary on the custom format? Are the semantics okay? I'll clarify a bit, but it's essentially the same semantics as `for`, so `lb < ub` and `step > 0`.